### PR TITLE
bug/ICL-171 onchain claim is not registered from mobile

### PR DESCRIPTION
**Pull request by JGiter**
https://energyweb.atlassian.net/browse/ICL-171

### CODE REVIEW

File: src/app/shared/services/login/login.service.ts [hunk 1]
```diff
@@ -45,11 +45,16 @@ export class LoginService {
   }
 
   isSessionActive() {
-    return (Boolean(localStorage.getItem(PROVIDER_TYPE)) && Boolean(localStorage.getItem(PUBLIC_KEY)));
+    return (
+      Boolean(localStorage.getItem(PROVIDER_TYPE)) 
+      && Boolean(localStorage.getItem(PUBLIC_KEY)) 
+      && Boolean(localStorage.getItem(IS_ETH_SIGNER))
```
Comment (dawidgil) [inline src/app/shared/services/login/login.service.ts:51]: is `IS_ETH_SIGNER` const used somewhere inside iam-client-lib? The same with `PUBLIC_KEY` ? It's more like a question, if we need to handle it inside library as this is for storing session.

Comment (JGiter) [reply src/app/shared/services/login/login.service.ts:51]: Yes, iam-lib uses both these. `IS_ETH_SIGNER` is used to correctly sign message and `PUBLIC_KEY` is needed by did-reg, but might be deprecated later

File: src/app/shared/services/iam.service.ts [hunk 2]
```diff
@@ -87,6 +87,10 @@ export class IamService {
     return this.signerService.providerType;
   }
 
+  get isEthSigner() {
+    return this.signerService.isEthSigner;
```
Comment (dawidgil) [inline src/app/shared/services/iam.service.ts:91]: Another question, can this be null or undefined or always is set to true/false ?

Comment (JGiter) [reply src/app/shared/services/iam.service.ts:91]: It should be defined as long as the session is active

**Review by dawidgil — COMMENTED** [review]
Approved. However tests are failing, so please fix it before merging.

File: src/app/shared/services/login/login.service.ts (hunk 1 above)
Comment (jrhender) [reply src/app/shared/services/login/login.service.ts:51]: @JGiter I think we can remove `PUBLIC_KEY` as we're no longer requiring that it be added to a DID document, right? We can do this in a separate story though

Comment (JGiter) [reply src/app/shared/services/login/login.service.ts:51]: @jrhender  That is right, but first we need to change `EwSigner` to not require public key

Comment (jrhender) [reply src/app/shared/services/login/login.service.ts:51]: Ah okay, makes sense, thanks @JGiter

**Review by dawidgil — APPROVED** [review]

**Pull request merged**